File: Models/RemoteActionResponse.cs
namespace YFRemote.Server.Models;

public sealed record RemoteActionResponse(
    bool Success,
    string? Error = null,
    string? RequestId = null)
{
    public static RemoteActionResponse Ok() => new(true);

    public static RemoteActionResponse Fail(string error) => new(false, error);
}

File: android/app/src/main/java/com/yfremote/android/server/models/RemoteActionResponse.kt
package com.yfremote.android.server.models

import kotlinx.serialization.Serializable

// Spiegelt Models/RemoteActionResponse.cs.
@Serializable
data class RemoteActionResponse(
    val success: Boolean,
    val error: String? = null,
    val requestId: String? = null,
) {
    companion object {
        fun ok() = RemoteActionResponse(success = true)

        fun fail(error: String) = RemoteActionResponse(success = false, error = error)
    }
}
